report_thesis/src/sections/background.tex

```diff
@@ -227,28 +227,33 @@ \subsubsection{Extra Trees Regressor (ETR)}
-$$
-    \hat{f}^{(b)}(\mathbf{x}) = \hat{f}^{(b-1)}(\mathbf{x}) + \lambda \hat{f}_b(\mathbf{x}),
-$$
-where $\hat{f}_b(\mathbf{x})$ is the prediction of the $b$-th tree and $\lambda$ is the learning rate. Residuals are updated as:
-$$
-    r_i^{(b)} = y_i - \hat{f}^{(b-1)}(\mathbf{x}_i),
-$$
-Each tree is then trained on these updated residuals.
-The repetitive process of fitting a weak learner to predict the residuals and using its predictions to update the model gives us the final model:
+Gradient Boosting is a machine learning technique used for various tasks, including regression and classification.
+The fundamental concept involves sequentially adding models to minimize a loss function, where each successive model addresses the errors of the ensemble of preceding models.
+
+This technique utilizes gradient descent to optimize the loss function, allowing for the selection of different loss functions depending on the specific task.
+\gls{gbr} is a specialized application of gradient boosting for regression tasks, where it minimizes a regression-appropriate loss function, such as mean squared error or mean absolute error.
+
+The process starts with an initial model $f_{0}(x)$ that minimizes the loss function over the entire dataset:
 $$
-    \hat{f}(\mathbf{x}) = \sum_{b=1}^{B} \lambda \hat{f}_b(\mathbf{x})
+f_{0}(x)=\arg\min_{\gamma}\sum^{N}_{i=1}L(y_{i},\gamma)
 $$
-In the context of regression, gradient boosting aims to minimize the difference between the predicted values and the actual target values by fitting successive trees to the residuals.
-To minimize errors, gradient descent is used to iteratively update model parameters in the direction of the negative gradient of the loss function, thereby following the path of steepest descent~\cite{gradientLossFunction}.
+where $L(y,\hat{y})$ is the chosen loss function, $N$ is the number of samples, $y_{i}$ are the true values, and $\gamma$ is a constant.
+
+Then we start the iterative process of adding models to the ensemble.
+For each iteration $m$, from $1$ to $M$:
+
+\begin{enumerate}
+    \item Compute the residuals of the current model. For regression, this could be the squared error loss, $L(y, \hat{y}) = (y - \hat{y})^2$. The residuals $r_{i}^{(m)}$ for each data point $i$ are calculated as $r_{i}^{(m)} = y_{i} - f_{m-1}(x_{i})$, where $f_{m-1}(x_{i})$ is the prediction of the previous model.
+    \item Fit a new model $h_{m}(x)$ to the residuals. This model aims to correct the errors of the current ensemble by using the residuals instead of ground truth values. Essentially, $h_{m}(x)$ tries to predict the residuals $r_{i}^{(m)}$.
+    \item Update the ensemble model by adding the predictions of the new model $h_{m}(x)$, multiplied by a learning rate $\eta$. The learning rate $\eta$ controls the contribution of each new model to the ensemble, preventing overfitting by scaling the updates:
+    $$
+    f_{m}(x)=f_{m-1}(x)+\eta h_{m}(x)
+    $$
```

Comment:
This is where I draw inspiration from Burkov's work. It much simplifies this part of the algorithm, and introduces the learning rate, which is present in modern implementations. To avoid confusion with the Hastie notation, I use $h$ to denote the function.

![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/9a89a6c5-f78b-412b-b359-95ca7afcf14b)